server/tests/booking.test.ts

```diff
@@ -0,0 +1,204 @@
+      office,
+      date,
+      parking: true,
+    };
+    const createResponse = await app
+      .post('/api/bookings')
+      .send(createBookingBody)
+      .set('bearer', normalUserEmail);
+
+    expect(createResponse.ok).toBe(true);
+    expect(Object.keys(createResponse.body)).toEqual([
+      'id',
+      'created',
+      'user',
+      'date',
+      'office',
+      'lastCancellation',
+      'parking',
+    ]);
+    expect(typeof createResponse.body?.id).toBe('string');
+    expect(createResponse.body).toMatchObject(createBookingBody);
+
+    const getCreatedBookingResponse = await app
+      .get(`/api/bookings?user=${normalUserEmail}`)
+      .set('bearer', normalUserEmail);
+    expect(getCreatedBookingResponse.body).toContainEqual(createResponse.body);
+
+    const getOfficeBookingsResponse = await app.get(`/api/offices`).set('bearer', normalUserEmail);
+    const officeData = getOfficeBookingsResponse.body.find((item: any) => item.name === office);
+    const slot = officeData.slots.find((item: any) => item.date === date);
+    expect(slot.booked).toEqual(1);
+    expect(slot.bookedParking).toEqual(1);
+
+    const deleteResponse = await app
+      .delete(`/api/bookings/${createResponse.body.id}`)
+      .set('bearer', normalUserEmail);
+    expect(deleteResponse.status).toBe(204);
+  });
+
+  test('can delete booking with parking and successfully increase booking count and parking count', async () => {
+    const office = officeQuotas[0].name;
+    const date = format(addDays(new Date(), 1), 'yyyy-MM-dd');
+    const createBookingBody = {
+      user: normalUserEmail,
+      office,
+      date,
+      parking: true,
+    };
+    const createResponse = await app
+      .post('/api/bookings')
+      .send(createBookingBody)
+      .set('bearer', normalUserEmail);
+
+    expect(createResponse.ok).toBe(true);
+    expect(Object.keys(createResponse.body)).toEqual([
+      'id',
+      'created',
+      'user',
+      'date',
+      'office',
+      'lastCancellation',
+      'parking',
+    ]);
+    expect(typeof createResponse.body?.id).toBe('string');
+    expect(createResponse.body).toMatchObject(createBookingBody);
+
+    const getCreatedBookingResponse = await app
+      .get(`/api/bookings?user=${normalUserEmail}`)
+      .set('bearer', normalUserEmail);
+    expect(getCreatedBookingResponse.body).toContainEqual(createResponse.body);
+
+    const deleteResponse = await app
+      .delete(`/api/bookings/${createResponse.body.id}`)
+      .set('bearer', normalUserEmail);
+    expect(deleteResponse.status).toBe(204);
+
+    const getOfficeBookingsResponse = await app.get(`/api/offices`).set('bearer', normalUserEmail);
+    const officeData = getOfficeBookingsResponse.body.find((item: any) => item.name === office);
+    const slot = officeData.slots.find((item: any) => item.date === date);
+    expect(slot.booked).toEqual(0);
+    expect(slot.bookedParking).toEqual(0);
+  });
+
+  test('cannot exceed quota', async () => {
```

Comment:
Don't think this is testing what you think it's testing - this is failing due to not being allowed to book the same office twice on the same day - not specifically due to quotas.

Probably worth having a test to try and book parking for an office without free parking and see it fail and roll-back correctly.